tidy3d/web/core/stub.py

```diff
@@ -0,0 +1,66 @@
+"""Stub class"""
+
+from enum import Enum
+
+from pydantic.v1 import BaseModel
+
+
+class TaskType(str, Enum):
+    FDTD = "FDTD"
+    NORMAL = "NORMAL"
+    MODE_SOLVER = "MODE_SOLVER"
+    HEAT = "HEAT"
+
+
+class Stub(BaseModel):
+
+    def from_file(self, file_path: str) -> "Stub":
```

Comment:
so I still think these should be `abstractmethods` (and stub should be ABC), my comment was that I think `validate_pre_upload()` could maybe just not be an abstractmethod, since we can make it optional to implement in subclasses.

Reply:
removed Stub